roles/st2chatops/tasks/main.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Install latest st2chatops package
+  become: yes
+  package:
+    name: st2chatops
+    state: latest
+  when: st2chatops_version == "latest"
+  tags: [st2chatops, skip_ansible_lint]
+
+- name: Install pinned st2chatops package
+  become: yes
+  package:
+    name: st2chatops={{ st2chatops_version }}
+    state: present
+  when: st2chatops_version != "latest"
+  tags: [st2chatops, skip_ansible_lint]
+
+- name: Add st2_api_key in st2chatops.env
+  become: yes
+  replace:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    regexp: '(?<=ST2_API_KEY=)(.*)$'
+    replace: '{{ st2chatops_config.ST2_API_KEY }}'
+  when: st2chatops_config.ST2_API_KEY != "CHANGE-ME-PLEASE"
+  register: st2chatops_api_key
+  tags: st2chatops
+
+- name: Comment Username, Password and Auth URL, if API_KEY provided
+  become: yes
+  replace:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    regexp: '{{ item }}'
+    replace: '# \1'
+  with_items:
+    - '^(export ST2_AUTH_URL.*)'
+    - '^(export ST2_AUTH_USERNAME.*)'
+    - '^(export ST2_AUTH_PASSWORD.*)'
+  when: st2chatops_api_key.changed
+  tags: [st2chatops, skip_ansible_lint]
+
+- name: Uncomment Hubot Slack Adapter in st2chatops.env
+  become: yes
+  replace:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    regexp: '^# (export HUBOT_ADAPTER=slack)$'
+    replace: '\1'
+  when: st2chatops_api_key.changed and st2chatops_config.HUBOT_ADAPTER == "slack"
+  register: uncomment_slack_adapter
+  tags: [st2chatops, skip_ansible_lint]
+
+- name: Uncomment and replace Hubot Slack Token in st2chatops.env
+  become: yes
+  replace:
+    dest: /opt/stackstorm/chatops/st2chatops.env
+    regexp: '^# (export HUBOT_SLACK_TOKEN.).*'
```

Comment:
>I don't even want to care about it, it's user's problem. 

No. We explicitly decided that we'll supply this in commented form in our config file so they don't have to go guess. If someone is adding support for an adapter here, they should include a .j2 file. A .j2 file is a schema for the adapter configuration. You won't have the following problem if you went with this approach.

> should we write that key:value in st2chatops.env too or maybe throw an error that such env var is not available. WDYT?

> Who knows about more vars hidden in the code?

Exactly the problem we should solve and not leave it to user

>we obviously won't include env vars in ansible-st2.

I'd validate the "adapter" variable against a set of supported adapters. A supported adapter means presence of .j2 file.